scripts/release.mjs

```diff
@@ -0,0 +1,104 @@
+import {execSync} from "node:child_process";
+import yargs from "yargs";
+import {hideBin} from "yargs/helpers";
+import inquirer from "inquirer";
+
+// Script to make releasing easier
+// Run with --help to see usage
+
+const cmd = (cmd, opts={}) => execSync(cmd, {encoding: "utf8", stdio: ["pipe", "pipe", "ignore"], ...opts}).trim();
+
+const argv = yargs(hideBin(process.argv))
+  .usage("Release lodestar")
+  .example([
+    ["$0 -t v0.36.0", "Release version 0.36.0 using the current commit"]
+  ])
+  .options({
+    tag: {
+      alias: "t",
+      demandOption: true,
+      type: "string",
+      describe: "The tag to release",
+    },
+    commit: {
+      alias: "c",
+      type: "string",
+      default: cmd("git rev-parse --short HEAD"),
+      describe: "The commit to tag",
+    },
+    yes: {
+      alias: "y",
+      type: "boolean",
+      describe: "Automatic yes to prompts"
+    }
+  })
+  .version("v0.1.0")
+  .alias("v", "version")
+  .alias("h", "help")
+  .strict()
+  .help()
+  .argv;
+
+// Validate the supplied version
+const versionCaptureRegex=/^(v[0-9]+\.[0-9]+)\.[0-9]+(-beta\.[0-9]+)?$/
+const versionMatch = versionCaptureRegex.exec(argv.tag);
+if (versionMatch == null) {
+  console.log(`Tag must match ${versionCaptureRegex}`);
+  process.exit(1);
+}
+
+const tag = argv.tag;
+const commit = argv.commit;
+const commitMessage = cmd(`git show-branch --no-name ${commit}`);
+// The branch is assumed from the tag
+const branch = `${versionMatch[1]}.x`;
+
+console.log("Tag", tag)
+console.log("Commit", commit, commitMessage)
+console.log("Branch", branch)
+
+// Ensure the branch exists
+try {
+  cmd(`git show-branch --no-name ${branch}`);
+} catch (e) {
+  console.log(`Branch ${branch} does not exist`);
+  process.exit(1);
+}
+
+// Ensure the commit exists in the branch (last 10 commits)
+const last10Commits = cmd(`git log --oneline -n 10 ${branch}`);
+const commitMatch = last10Commits.match(commit);
+if (commitMatch == null) {
+  console.log(`Commit ${commit} does not belong to branch ${branch}`);
+  process.exit(1);
+}
+
+// Last chance to exit
+if (!argv.yes) {
+  const input = await inquirer.prompt([
+    {
+      name: "yes",
+      type: "confirm",
+      message: "Do you want to proceed?",
+    },
+  ]);
+  if (!input.yes) {
+    process.exit(1);
+  }
+}
+
```

Comment:
how do we prevent this error `TAG v0.36.1-beta.0 does not include LOCAL_VERSION 0.36.0` from https://github.com/ChainSafe/lodestar/runs/6476719731?check_suite_focus=true ? 

if yes, we should do a check here before we create a tag.

Reply:
That's a good idea! This condition should be checked here too

Reply:
The script now calls `lerna version $TAG` if the tag does not match `lerna.json` (which bumps the versions in package.jsons, commits, tags and pushes).
Otherwise, if the tag matches `lerna.json`, it tags and pushes manually with git.